x-pack/plugins/ml/public/application/components/annotations/annotation_flyout/index.tsx

```diff
@@ -150,11 +169,22 @@ class AnnotationFlyoutUI extends Component<CommonProps & Props> {
   };
 
   public saveOrUpdateAnnotation = () => {
-    const { annotation } = this.props;
-
-    if (annotation === null) {
+    const { annotation: originalAnnotation, chartDetails, detectorIndex } = this.props;
+    if (originalAnnotation === null) {
       return;
     }
+    const annotation = cloneDeep(originalAnnotation);
+
+    if (this.state.applyAnnotationToSeries && chartDetails?.entityData?.entities) {
+      chartDetails.entityData.entities.forEach((entity: Entity) => {
+        const { fieldName, fieldType, fieldValue } = entity;
+        annotation[getAnnotationFieldName(fieldType)] = fieldName;
+        annotation[getAnnotationFieldValue(fieldName)] = fieldValue;
```

Comment:
Fixed in https://github.com/elastic/kibana/pull/70198/commits/a56fa66e1f32c8a9bb083bb7e200c7a12b66e77e

Reply:
This seems to work ok now when creating an annotation, but I got this error when trying to edit an annotation for a series with a `by` and `over` field:

![image](https://user-images.githubusercontent.com/7405507/86465421-7d32a000-bd29-11ea-9d5e-aa9bcf5c449e.png)

Reply:
Adding and editing annotations should work correctly now in https://github.com/elastic/kibana/pull/70198/commits/1f2684dbbc91ca4a7d83ab93d8d4cfb79a5180c2. There was also a bug previously where it will update the annotation with not all the fields because it was only getting the annotation coming from the table row, which does not have all the information.

Editing a system generated annotation will keep the `event` field as is https://github.com/elastic/kibana/pull/70198/commits/e57889f722de503ae270b0a3a84bd65504394792.